alibi/explainers/tests/test_simiarlity/test_backends.py

```diff
@@ -41,3 +42,90 @@ def test_backends(random_cls_dataset, linear_models):
     torch_grads = np.sort(torch_grads)
     tf_grads = np.sort(tf_grads)
     np.testing.assert_allclose(torch_grads, tf_grads, rtol=1e-04)
+
+
+@pytest.mark.parametrize('trainable_emd, grads_shape', [(True, (61, )), (False, (21, ))])
+def test_tf_embedding_similarity(trainable_emd, grads_shape):
+    """Test GradSim explainer correctly handles sparcity and non-trainable layers for tensorflow.
+
+    Test that `tensorflow` embedding layers work as expected and also that layers
+    marked as non-trainable are not included in the gradients.
+    See https://github.com/SeldonIO/alibi/issues/828.
+    """
+    model = tf.keras.models.Sequential([
+        tf.keras.layers.Embedding(10, 4, input_shape=(5,), trainable=trainable_emd),
+        tf.keras.layers.Flatten(),
+        tf.keras.layers.Dense(1)
+    ])
+
+    X = tf.random.uniform(shape=(1, 5), minval=0, maxval=10, dtype=tf.float32)
+    Y = tf.random.uniform(shape=(1, 1), minval=0, maxval=10, dtype=tf.float32)
+    loss_fn = tf.keras.losses.MeanSquaredError()
+    tf_grads = _TensorFlowBackend.get_grads(model, X, Y, loss_fn)
+    assert tf_grads.shape == grads_shape  # (4 * 10) + (5 * 4) + 1
```

Comment:
nitpick: perhaps change to `(4 * 10) * trainable_emd + ...` to emphasize that the first term may be absent so the equation is correct for both tests